File: CardOverflow.Server/Gulid.cs
using NUlid;
using System;
using System.Collections.Generic;
using System.Linq;

namespace CardOverflow.Server {
  public static class Gulid {
    
    public static Guid Create() =>
      Ulid.NewUlid().ToGuid();

    public static List<Guid> Create(int count) =>
      Enumerable.Range(0, count).Select(_ => Create()).ToList();
  
  }
}

File: CardOverflow.Api/Gulid.fs
namespace CardOverflow.Api

open CardOverflow.Debug
open MappingTools
open CardOverflow.Entity
open CardOverflow.Pure
open CardOverflow.Pure.Core
open System
open System.Linq
open FsToolkit.ErrorHandling
open System.Security.Cryptography
open System.Text
open System.Collections.Generic
open System.Text.RegularExpressions
open System.Collections
open NUlid
open NodaTime

type Ulid =
    static member create with get () = Ulid.NewUlid().ToGuid()
    static member infinite with get () = Seq.initInfinite (fun _ -> Ulid.create)
    static member resizeList i xs = 
        List.resize i Ulid.infinite xs
    static member createMany i = [1..i] |> List.map (fun _ -> Ulid.create)

type UpsertIdsModule =
    static member create with get () =
        {   ConceptId = Ulid.create
            ExampleId = Ulid.create
            LeafId = Ulid.create
            CardIds = [ Ulid.create ] }
